R/display.R

```diff
@@ -23,52 +23,120 @@ display1 <- function(mimetype, content, metadata=NULL) {
     base_display(data, metadata)
 }
 
-#'Display HTML output
+prepare_content <- function(data = NULL, file = NULL, isbinary = TRUE) {
+    if (is.null(file) == is.null(data))
+        stop('Either need to specify data or file, but not both')
+    
+    if (isbinary) {
+        if (!is.null(file)) {
+            content <- base64encode(file)
+        } else if (is.raw(data)) {
+            content <- base64encode(data)
+        } else stop('Data needs to be a raw vector')
+    } else {
+        if (!is.null(file)) {
+            content <- readChar(file, file.info(file)$size)
+        } else if (is.character(data)) {
+            content <- data
+        } else stop('Data needs to be a character vector')
+    }
+    
+    content
+}
+
+display_raw <- function(mimetype, isbinary, data, file, metadata = NULL) {
```

Comment:
Parameter ordering seems a bit inconsistent - prepare_content takes (data, file, isbinary), while display_raw has isbinary before data or file. Is there a reason to do it like this?